src/components/BitcoinAmountInput.tsx

```diff
@@ -129,10 +129,12 @@ const BitcoinAmountInput = forwardRef(
                 )
                 return
               } else {
-                const value: number = valueOrNan
-
+                const value: number = floatValueOrNan
                 let numberValues: string | undefined
-                const unit = unitFromValue(String(value))
+                const unit =
+                  valueOrNan.includes('.') && parseFloat(valueOrNan)
```

Comment:
Does this also work for languages where `.` is not the separator? e.g. afaik in italian it is `,` ..

Reply:
If not, personally, I think it is not _that_ important. Internationalization support of number input field is quite cumbersome.

Reply:
> Does this also work for languages where `.` is not the separator? e.g. afaik in italian it is `,` ..

I may be wrong but BitcoinAmountInput only takes `numbers`, `e` & `.` as input, so i implemented on the basis of that
and also function `unitFromValue` identifies if the user input is in `sata` or `BTC` on the basis of `include('.')`

Reply:
> > Does this also work for languages where `.` is not the separator? e.g. afaik in italian it is `,` ..
> 
> I may be wrong but BitcoinAmountInput only takes `numbers`, `e` & `.` as inputs, so i implemented on the basis of that and also function `unitFromValue` identifies if the user raw input is in `sata` or `BTC` on the basis of `include('.')`

Yeah, it can be tricky to get it right across different browser implementations. If language settings are changed, Firefox lets me input `1,01` while Chromium does not.

Reply:
Oh, now I get it. I know it's not that important, but from a user's perspective, it might be seen as a bug in the input field.

Reply:
![image](https://github.com/user-attachments/assets/764be3ac-92a8-4cc4-9d7a-4d0277822ffb)
Hey, I tried what you suggested and guess what? It is indeed actually working. You were right, the `,` is converted into `.`. So Ig that solves our problem.

Reply:
Would it be possible to write test cases to verify this behaviour? Or is the existing code too clunky for that?